pkg/arvo/tests/sys/zuse/html.hoon

```diff
@@ -89,4 +90,745 @@
+      %+  expect-eq
+        !>  "[2,2,2]"
+        !>  (en-json [%a ~[two two two]])
+    ==
+  ++  test-en-json-objects
+    :: opening curly braces are escaped to avoid urbit string literal
+    :: interpolation
+    ;:  weld
+      :: empty object
+      %+  expect-eq
+        !>  "\{}"
+        !>  (en-json [%o ~])
+      :: one property
+      %+  expect-eq
+        !>  "\{\"foo\":2}"
+        !>  (en-json [%o (molt ~[['foo' two]])])
+      :: multiple properties are comma-separated
+      %+  expect-eq
+        !>  "\{\"foo\":2,\"bar\":true}"
+        !>  (en-json [%o (molt ~[['foo' two] ['bar' tru]])])
+      :: object keys use same encoding logic as strings
+      %+  expect-eq
+        :: {"\u0007\"\n\\":true}
+        !>  "\{\"\\u0007\\\"\\n\\\\\":true}"
+        !>  (en-json [%o (molt ~[['\07"\0a\\' tru]])])
+    ==
+  ++  test-en-json-complex-structure
+    %+  expect-eq
+      !>  "[\{},4,[[],[\{\"foo\":\{\"4\":4,\"true\":true}}]]]"
+      !>  (en-json [%a ~[[%o ~] [%n '4'] [%a ~[[%a ~] [%a ~[(frond:enjs:format 'foo' (pairs:enjs:format ~[['4' [%n '4']] ['true' [%b &]]]))]]]]]])
+  ++  test-de-json-simple-values
+    =,  html
+    ;:  weld
+      %+  expect-eq
+        !>  `~
+        !>  (de-json 'null')
+      %+  expect-eq
+        !>  `[%b &]
+        !>  (de-json 'true')
+      %+  expect-eq
+        !>  `[%b |]
+        !>  (de-json 'false')
+    ==
+  :: The following parser test suite (test-de-json-bad-examples and
+  :: test-en-json-suite) is adapted from https://github.com/nst/JSONTestSuite/
+  ::
+  ::
+  :: MIT License
+  ::
+  :: Copyright (c) 2016 Nicolas Seriot
+  ::
+  :: Permission is hereby granted, free of charge, to any person
+  :: obtaining a copy of this software and associated documentation
+  :: files (the "Software"), to deal in the Software without
+  :: restriction, including without limitation the rights to use, copy,
+  :: modify, merge, publish, distribute, sublicense, and/or sell copies
+  :: of the Software, and to permit persons to whom the Software is
+  :: furnished to do so, subject to the following conditions:
+  ::
+  :: The above copyright notice and this permission notice shall be
+  :: included in all copies or substantial portions of the Software.
+  ::
+  :: THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+  :: KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+  :: WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+  :: NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+  :: BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+  :: ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+  :: CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+  :: SOFTWARE.
+  ::
+  ::
+  :: These are all inputs that should be rejected by a valid json parser
+  ++  test-de-json-bad-examples
+    %-  zing
+    %-  turn
+    :_  |=  [exmpl=@t name=tape]
+          ^-  tang
```

Comment:
This indentation is off, the `^-` and subsequent lines should be aligned with the `|=`.